dev-tools/omdb/src/bin/omdb/db.rs

```diff
@@ -1743,32 +1762,54 @@ async fn cmd_db_eips(
         }
     }
 
-    #[derive(Tabled)]
     enum Owner {
-        Instance { project: String, name: String },
-        Service { kind: String },
+        Instance { id: Uuid, project: String, name: String },
+        Service { id: Uuid, kind: String },
+        Project { id: Uuid, name: String },
         None,
     }
 
-    impl Display for Owner {
-        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    impl Owner {
+        fn kind(&self) -> &'static str {
             match self {
-                Self::Instance { project, name } => {
-                    write!(f, "Instance {project}/{name}")
+                Owner::Instance { .. } => "instance",
+                Owner::Service { .. } => "service",
+                Owner::Project { .. } => "project",
+                Owner::None => "none",
+            }
+        }
+
+        fn id(&self) -> String {
+            match self {
+                Owner::Instance { id, .. }
+                | Owner::Service { id, .. }
+                | Owner::Project { id, .. } => id.to_string(),
+                Owner::None => "none".to_string(),
+            }
+        }
+
+        fn description(&self) -> String {
+            match self {
+                Self::Instance { project, name, .. } => {
+                    format!("{project}/{name}")
                 }
-                Self::Service { kind } => write!(f, "Service {kind}"),
-                Self::None => write!(f, "None"),
+                Self::Service { kind, .. } => kind.to_string(),
+                Self::Project { name, .. } => name.to_string(),
+                Self::None => "none".to_string(),
             }
         }
     }
 
     #[derive(Tabled)]
+    #[tabled(rename_all = "SCREAMING_SNAKE_CASE")]
     struct IpRow {
         ip: ipnetwork::IpNetwork,
         ports: PortRange,
-        kind: String,
+        kind: IpKind,
         state: IpAttachState,
-        owner: Owner,
+        owner_kind: &'static str,
+        owner_id: String,
+        owner_description: String,
```

Comment:
I would probably change this to `owner_name`. Description, to me, implies the random text that end users can tack onto named resources.

Reply:
:+1: fixed in [ff4889a](https://github.com/oxidecomputer/omicron/pull/5064/commits/ff4889a232434340fc1c0dbf8b2ee8b1bc886aeb)